### PR TITLE
Chore: error while run command against wrong cluser

### PR DESCRIPTION
**Pull request by ycyaoxdu**
prompt warnings while running either hub oriented command against managedcluster or managedcluster oriented command against hub

Signed-off-by: ycyaoxdu <yaoyuchen0626@163.com>

### CODE REVIEW

**Comment by ycyaoxdu** [comment]
/assign @qiujian16  
PTAL

**Comment by ycyaoxdu** [comment]
related issue: #133

File: pkg/cmd/create/work/exec.go [hunk 1]
```diff
@@ -34,6 +36,16 @@ func (o *Options) complete(cmd *cobra.Command, args []string) (err error) {
...
 func (o *Options) validate() (err error) {
+	config, err := o.ClusteradmFlags.KubectlFactory.ToRESTConfig()
+	if err != nil {
+		return err
+	}
+	client, err := operatorclient.NewForConfig(config)
+	if err != nil {
+		return err
+	}
+	warning.CheckKlusterletAbsent(client)
```
Comment (qiujian16) [inline pkg/cmd/create/work/exec.go:47]: I think it should validate hub.

File: pkg/cmd/delete/clusterset/exec.go [hunk 2]
```diff
@@ -21,12 +23,23 @@ func (o *Options) complete(cmd *cobra.Command, args []string) (err error) {
...
 func (o *Options) validate() (err error) {
+	config, err := o.ClusteradmFlags.KubectlFactory.ToRESTConfig()
+	if err != nil {
+		return err
+	}
+	client, err := operatorclient.NewForConfig(config)
+	if err != nil {
+		return err
+	}
+	warning.CheckKlusterletAbsent(client)
```
Comment (qiujian16) [inline pkg/cmd/delete/clusterset/exec.go:34]: validate hub

**Review by qiujian16 — COMMENTED** [review]
I do not think it should be just a warning, the command will fail, right?

Also could we have a validate func in genericControlPlane to avoid redundant code.

File: go.mod [hunk 3]
```diff
@@ -79,6 +79,7 @@ require (
 	github.com/josharian/intern v1.0.0 // indirect
 	github.com/json-iterator/go v1.1.12 // indirect
 	github.com/kr/pretty v0.2.1 // indirect
+	github.com/kubernetes-sigs/kubebuilder v1.0.8 // indirect
```
Comment (qiujian16) [inline go.mod:82]: where do we import this?

Comment (ycyaoxdu) [reply go.mod:82]: wrong import while fix integration test, will remove

**Pull request merged**